src/System.Management.Automation/namespaces/FileSystemProvider.cs

```diff
@@ -2069,7 +2069,7 @@ public static string NameString(PSObject instance)
                 {
                     if (InternalSymbolicLinkLinkCodeMethods.IsReparsePointLikeSymlink(fileInfo))
                     {
-                        return $"{PSStyle.Instance.FileInfo.SymbolicLink}{fileInfo.Name}{PSStyle.Instance.Reset} -> {InternalSymbolicLinkLinkCodeMethods.GetTarget(instance)}";
+                        return $"{PSStyle.Instance.FileInfo.SymbolicLink}{fileInfo.Name}{PSStyle.Instance.Reset} -> {fileInfo.LinkTarget}";
```

Comment:
It would be great to get rid of IsReparsePointLikeSymlink() call and check `fileInfo.LinkTarget is not null` but .Net API has slightly different behavior - can we enhance the .Net API?

https://github.com/PowerShell/PowerShell/blob/176303d172bcb0726e097ac5139de9f724ecec6b/src/System.Management.Automation/namespaces/FileSystemProvider.cs#L8301-L8308

(Also you can see as IsReparsePointLikeSymlink() is used in other places. It seems it work the same for directory removal scenario but directory enumeration requires enhaсing of .Net API)